src/app/page.tsx

```diff
@@ -1,36 +1,35 @@
 import Link from "next/link";
 
-import { LatestPost } from "~/app/_components/post";
-import { getServerAuthSession } from "~/server/auth";
 import { api, HydrateClient } from "~/trpc/server";
-import { UserProfile } from "./_components/userProfile/UserProfile";
+import { getServerAuthSession } from "~/server/auth";
+import UserProfile from "~/app/_components/userProfile/UserProfile";
 
 export default async function Home() {
-  const hello = await api.post.hello({ text: "from tRPC" });
-  const session = await getServerAuthSession();
-
-  void api.post.getLatest.prefetch();
+  const SESSION = await getServerAuthSession();
 
-  if (!session) {
+  if (!SESSION) {
     return (
       <div>
         <p>
-          Welcome to <code>trpc</code> with <code>next-auth</code>!
-        </p>
-        <p>
-          <Link href="/api/auth/signin">Sign in</Link> to see the latest post.
+          <Link href="/api/auth/signin">Sign in</Link>
         </p>
       </div>
     );
   }
 
+  const TEAM = await api.team.getByUserId({userId: SESSION.user.id})
+
   return (
     <HydrateClient>
       <div>
         <p>
-          Welcome back, <code>{session.user.email}</code>!
+          Welcome back, <code>{SESSION.user.email}</code>!
         </p>
-        <UserProfile session={session} />
+        {TEAM ? (
+          <UserProfile SESSION={SESSION} TEAM={TEAM}/>
```

Comment:
I think it would be best if the UserProfile didn't take any props and just called getServerAuth session. Then we can wrap the whole UserProfile in a Suspense later. 

Let's try to keep data fetching within each respective component and avoid passing props unless necessary.